### ISSUE TITLE
Video player plugin should support WebVTT closed captions file format. 

### ISSUE THREAD

**Issue by johnsonmh** [flutter/flutter#50595]
## Use case

This file format is commonly used on the web. https://github.com/flutter/flutter/issues/25388 was resolved, but the only supported file type right now is SubRip. More specifically, when the `video_player` plugin is used to display [HLS](https://en.wikipedia.org/wiki/HTTP_Live_Streaming), we want the ability to read captions from the stream and display them in our app. 

Example HLS with captions: https://developer.apple.com/streaming/examples/advanced-stream-fmp4.html

## Proposal

We should write a subclass of `ClosedCaptionFile` in the `video_player` plugin that can read closed caption files in WebVTT format. We also _may_ need to modify the player to support reading captions on the go, as a stream while the video file is being streamed.

**Comment by snaeji** [comment]
If it's any help i modified the 0.10.1+3 version of video_player plugin and chewie to support HLS embedded and WebVTT subtitles. It's a raw solution but working great in our production app.

And it supports subtitles on Android only.

https://github.com/snaeji/video_player-0.10.1-3

**Comment by smileypop** [comment]
I agree the video player should support WebVTT as well as streaming directly from an online source.

[Example](https://www.youtube.com/api/timedtext?lang=en&v=Mt5C5fhuU_0&fmt=vtt&name=https://www.youtube.com/api/timedtext?lang=en&v=Mt5C5fhuU_0&fmt=vtt&name=)

**Comment by johnsonmh** [comment]
Just an update: this feature request is not on our roadmap at the moment, so it is a good candidate for any contributors interested in submitting a PR!

The PR would resemble https://github.com/flutter/plugins/pull/2473 (which added support for a different file format).

WebVTT parsing guidelines here: https://www.w3.org/TR/webvtt1/#file-parsing

**Comment by abhivinny** [comment]
@johnsonmh If this is not on the roadmap, what does Flutter team suggest in this case, to have video player implementation with CC done at the platform-specific sides? Also, it's hard to believe that WebVTT support is not on roadmap considering that's one of the core features of video players.

**Comment by JohnGalt1717** [comment]
Note also that this should be integrated into work for AES, Widevine etc. encryption because the 2 are related to eachother on iOS because of the approach that has to be taken.

**Comment by johnsonmh** [comment]
@ferrazrx That is awesome! Taking a look now

**Comment by StreamProjectsPeter** [comment]
<track> ideally should be implemented in full allowing VTT support which is essential for our work (multi platform tech presentations+search ability). It should be included for accessibility reasons in any case.   Also preview thumbnails.

**Comment by nateshmbhat** [comment]
Any updates on this?

**Comment by tech-andgar** [comment]
> Any updates on this?

FWIW, I think they forgot closed this case because already implemented in [v2.2.5](https://pub.dev/packages/video_player/changelog#225)

**Comment by stuartmorgan-g** [comment]
Fixed it https://github.com/flutter/plugins/pull/2878, the issue just didn't get auto-closed.
